scripts/build/termux_step_get_dependencies_python.sh

```diff
@@ -30,6 +30,9 @@ termux_step_get_dependencies_python() {
 
 		# adding and setting values ​​to work properly with python modules
 		export PYTHONPATH=$TERMUX_PYTHON_HOME/site-packages
+		if [ "$TERMUX_ON_DEVICE_BUILD" = "false" ]; then
+			export PYTHONPATH+=":$TERMUX_PYTHON_CROSSENV_PREFIX/build/lib/python${TERMUX_PYTHON_VERSION}/site-packages"
```

Comment:
This will mess up the target python. Can you give a case why this PR is needed?

Reply:
> This will mess up the target python.

Could you explain this in more detail?

> Can you give a case why this PR is needed?

I'm trying to rebuild some glibc packages with new `python-glibc` (3.12.6), and I encountered a problem with `gobject-introspection-glibc` package. The problem is that `gobject-introspection-glibc` package works with `python-glibc` directly, and it started requiring `setuptools` module in new `python-glibc`. The module can be installed via `termux_step_get_dependencies_python` function, but `python-glibc` doesn't see it, because it is installed for `python-crossenv`. So I think it would be good if `python-glibc` could find modules from `python-crossenv`. Here is the PR (it already tested with this change) - https://github.com/termux-pacman/glibc-packages/pull/287

Reply:
> Could you explain this in more detail?

`crossenv` uses virtualenv to simulate a python interceptor for the target python. If `PYTHONPATH` is set, then the simulated python will search it first, and some packages installed in the cross-virtualenv will never get searched -- packages in the build-virtualenv will be used instead, and this may lead to some issues. 

If `crossenv` is used, the `python` command should be redirected to  `$TERMUX_PYTHON_CROSSENV_PREFIX/cross/bin/python`. It seems that `$PREFIX/bin/python` is used in the build script of `gobject-introspection-glibc` (https://github.com/termux-pacman/glibc-packages/blob/ac3a9354b29a20dcff8f33740e564dcffa582612/gpkg/gobject-introspection/build.sh#L22)

If it is really necessay for `python-glibc`, you can guard it with `"$TERMUX_ON_DEVICE_BUILD" = "false" && "${TERMUX_PACKAGE_LIBRARY}" = "glibc"`.

Reply:
Ok, thanks for the explanation. I took this into account and made some changes in the `python-glibc` package. There I added a special variable `TERMUX_PYTHON_MAINPATH`, which works like the `PYTHONPATH` variable. But when using this new variable, the `PYTHONPATH` variable will be ignored.

Reply:
> crossenv uses virtualenv to simulate a python interceptor for the target python. If PYTHONPATH is set, then the simulated python will search it first, and some packages installed in the cross-virtualenv will never get searched -- packages in the build-virtualenv will be used instead, and this may lead to some issues.

One question: what's the problem with first specifying the path to the python target modules in the `PYTHONPATH` variable (so that python crossenv would check it first), and then specifying the path to the python crossenv modules (which is already included in python crossenv by default)? It's just that in python, the module search occurs by a list, and we can easily customize this list by specifying which paths need to be checked first.

Reply:
Scripts of some pacakges uses path like `$PYTHONPATH/site-packages` or `$PYTHON_SITE_PKG`. It would be better not modify them.